scilpy/image/resample_volume.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+
+from dipy.align.reslice import reslice
+import nibabel as nib
+import numpy as np
+
+
+def _interp_code_to_order(interp_code):
+    orders = {'nn': 0, 'lin': 1, 'quad': 2, 'cubic': 3}
+    return orders[interp_code]
+
+
+def resample_volume(img, ref=None, res=None, iso_min=False, interp='lin',
+                    enforce_dimensions=False):
+    """
+    Script to resample a dataset to match the resolution of another
+    reference dataset or to the resolution specified as in argument.
+    One of the following options must be chosen: ref, res or iso_min.
+
+    Parameters
+    ----------
+    img: nib.Nifti1Image
+        Image to resample.
+    ref:
+        Reference volume to resample to. This method is used only if ref is not
+        None. (default: None)
+    res: float, optional
+        Resolution to resample to. If the value it is set to is Y, it will
+        resample to an isotropic resolution of Y x Y x Y. This method is used
+        only if res is not None. (default: None)
+    iso_min: bool, optional
+        If true, resample the volume to R x R x R with R being the smallest
+        current voxel dimension. If false, this method is not used.
+    interp: str, optional
+        Interpolation mode. 'nn' = nearest neighbour, 'lin' = linear,
+        'quad' = quadratic, 'cubic' = cubic. (Default: linear)
+    enforce_dimensions: bool, optional
+        If True, enforce the reference volume dimension (only if res in not None).
+        (Default = False)
+
+    Returns
+    -------
+    resampled_image: nib.Nifti1Image
+        Resampled image.
+    """
+    data = img.get_data()
+    affine = img.get_affine()
+    original_zooms = img.get_header().get_zooms()[:3]
+
+    if ref is not None:
+        ref_img = nib.load(ref)
+        new_zooms = ref_img.header.get_zooms()[:3]
+    elif res is not None:
+        new_zooms = [res] * 3
+    elif iso_min:
+        min_zoom = min(original_zooms)
+        new_zooms = (min_zoom, min_zoom, min_zoom)
+    else:
```

Comment:
ref/ref/iso_min should be mutually exclusive, can you check to make sure there are not multiple options?